src/ioc/Ioc.java

```diff
@@ -1,18 +1,26 @@
 package ioc;
 
+import enums.IocServicesEnum;
 import exceptions.ioc.IocDuplicatedKeyException;
 import exceptions.ioc.IocKeyNotFoundException;
 import exceptions.ioc.IocKeyNullPointerException;
 import exceptions.ioc.IocValueNullPointerException;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
+import services.implementation.ClientServiceImpl;
 
 public class Ioc {
   private static Ioc instance;
   private final Map<String, Object> instanceMap = new HashMap<>();
 
-  private Ioc() {}
+  private Ioc() {
+    this.insertServicesInTheMap(this.instanceMap);
+  }
+
+  private void insertServicesInTheMap(Map<String, Object> instanceMap) {
```

Comment:
Rename this function to `registerServices`

Reply:
solved